### PR TITLE
feature(motor-control): add error message to move response varient

### PR DESCRIPTION
**Pull request by ryanthecoder**
[RET-1250](https://opentrons.atlassian.net/browse/RET-1250) firmware side work
Allows the motor controller to respond to moves with an error message in addition to the ack

### CODE REVIEW

File: include/can/core/ids.hpp [hunk 1]
```diff
@@ -125,9 +125,12 @@ enum class ErrorCode {
     invalid_byte_count = 0x3,
     invalid_input = 0x4,
     hardware = 0x5,
+    timeout = 0x6,
+    estop_detected = 0x7,
```
Comment (Laura-Danielle) [inline include/can/core/ids.hpp:129]: question for the future. Are we planning to also have subset error codes that don't apply to every microcontroller (i.e. sensor errors)

**Review by Laura-Danielle — APPROVED** [review]
This LGTM, had a future question and one comment about casting.

File: include/can/core/messages.hpp [hunk 2]
```diff
@@ -75,14 +75,14 @@ static auto add_resp_ind(Resposne& resp, const Request& req) -> void {
...
-    uint16_t error_code;
+    ErrorSeverity severity;
+    ErrorCode error_code;
 
     template <bit_utils::ByteIterator Output, typename Limit>
     auto serialize(Output body, Limit limit) const -> uint8_t {
         auto iter = bit_utils::int_to_bytes(message_index, body, limit);
-        iter = bit_utils::int_to_bytes(severity, iter, limit);
-        iter = bit_utils::int_to_bytes(error_code, iter, limit);
+        iter = bit_utils::int_to_bytes(uint16_t(severity), iter, limit);
```
[the remark this answers is not in the document]
Comment (ryanthecoder) [reply include/can/core/messages.hpp:84]: eh you're right I should use static here. Did some googling the difference is this:
c-style casts can either use dynamic or static cast depending on implementation so you're don't know which one gets used
static casts are always typed checked at compile time so they're considered safer.

[the remark this answers is not in the document]
Comment (sfoster1) [reply include/can/core/messages.hpp:84]: fun fact that doesn't matter: this is not actually a c-style cast. this invokes the [converting constructor](https://en.cppreference.com/w/cpp/language/converting_constructor) of `uint16_t`. a c-style cast would be `(uint16_t)severity`. i have a slight bias towards using static cast but it doesn't matter.

**Review by sfoster1 — APPROVED** [review]
Looks good to me.

**Pull request merged**